GenericDiffractometer.py

```diff
@@ -283,6 +290,36 @@ def init(self):
                 self.use_sc = True
 
 
+        # centring motors
+        self.centring_motors_list = ["phiz", "phiy","sampx","sampy"]
+	try:
+           self.use_sample_centring = self.getProperty("sample_centring")
```

Comment:
Yes, I noticed that. But since there is already a sample_centring hwobj and also ESRF uses it as well, so I keep it in the GenericDiffractometer to make it simpler for others. As you saw in the PR, I added a flag to make it compatible with your side as well.

In practice, you don’t need to do anything… It should work with the EMBLMiniDiff. It’s only if one want to use the sample_centring hwobj, one needs to configure it in the xml file as below

<sample_centring>True</sample_centring>



From: Ivars Karpics [mailto:notifications@github.com]
Sent: Thursday, April 14, 2016 1:28 PM
To: mxcube/HardwareObjects
Cc: Jie Nan
Subject: Re: [mxcube/HardwareObjects] Improvement of Genericdiffractometer - 2nd run (#33)


In GenericDiffractometer.py<https://github.com/mxcube/HardwareObjects/pull/33#discussion_r59701209>:

> @@ -283,6 +290,36 @@ def init(self):

>                  self.use_sc = True

>

>

> +        # centring motors

> +        self.centring_motors_list = ["phiz", "phiy","sampx","sampy"]

> +     try:

> +           self.use_sample_centring = self.getProperty("sample_centring")

From my point of view this should not be in GenericDiffractometer, because we use different centering algorithm (CentringMath) which I import in EMBLMiniDiff

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/mxcube/HardwareObjects/pull/33/files/6aeb0b483ec493fa92a3f27336915340e0594579#r59701209>